File: backend/models/Order.js
import mongoose from 'mongoose';

const orderItemSchema = new mongoose.Schema(
  {
    product: { type: mongoose.Schema.Types.ObjectId, ref: 'Product', required: true },
    quantity: { type: Number, required: true },
    price: { type: Number, required: true },
  },
  { _id: false }
);

const orderSchema = new mongoose.Schema(
  {
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    items: [orderItemSchema],
    total: { type: Number, required: true },
    status: {
      type: String,
      enum: ['pending', 'paid', 'shipped', 'delivered'],
      default: 'pending',
    },
    shippingAddress: {
      fullName: String,
      address: String,
      city: String,
      postalCode: String,
      country: String,
    },
    paymentIntentId: { type: String },
  },
  { timestamps: true }
);

const Order = mongoose.model('Order', orderSchema);

export default Order;

File: backend/app/models/order.rb
# frozen_string_literal: true

class Order < ApplicationRecord
  belongs_to :user

  validates :status, presence: true
  validates :line_items, presence: true

  enum status: { pending: 'pending', paid: 'paid', shipped: 'shipped' }, _default: 'pending'

  def as_json(options = {})
    super({
      methods: :_id,
      include: { user: { only: %i[id name email], methods: :_id } },
      except: [:updated_at]
    }.merge(options)).merge(
      'items' => line_items,
      'total' => total_amount.to_f,
      'createdAt' => created_at&.iso8601,
      'shippingAddress' => shipping_address
    )
  end
end
